data_analysis/utils.py

```diff
@@ -140,18 +140,20 @@ def calculate_distance_lat_lon(latlon_a: Tuple[float, float],
     return distance
 
 
-def map_ts_to_day_beginning_at_5am(timestamp: pd.Timestamp | str) -> datetime.date:
+def map_ts_to_interval_start(timestamp: pd.Timestamp | str, interval_start: 5, interval_length_h: 24) -> datetime.date:
     """
     Instead of considering a 'day' as ranging from 0.00-23.59, we will now
-    view it as 5.00-4.59. This function is to map a timestamp to a day, since
-    we cannot do "timestamp.date()" anymore.
+    view it as starting at 'interval_start' with length 'interval_length' in hours. This function is to map a timestamp to
+    an interval, since we cannot do "timestamp.date()" anymore.
     :param timestamp: an object containing date and time information.
-    :return: the day (from 5.00AM to 4.59AM) to which the timestamp is mapped.
+    :param interval_start: hour at which the interval starts
+    :param interval_length_h: length of the interval ('day')
```

Comment:
Meer een vraag, maar ik zie niet helemaal wanneer het nodig is om een ander interval dan een dag te gebruiken

Reply:
soms zou je de analyse misschien per week/maand/jaar willen doen